sssd_test_framework/utils/sss_override.py

```diff
@@ -0,0 +1,420 @@
+
+    def group(self, group: str) -> SSSOverrideGroup:
+        """
+        Get local override group object.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create local override
+                client.sss_override.group("users").add(name="o-users", gid=10001)
+
+                # Assert the overridden name and gid
+                assert client.tools.getent.group("o-users").gid == 10001
+
+        :param group: Group name.
+        :type group: str
+        :return: New group object.
+        :rtype: SSSOverrideGroup
+        """
+        return SSSOverrideGroup(self, group)
+
+
+class SSSOverrideUser:
+    """
+    Management of local override for users using sss_override.
+    """
+
+    def __init__(self, util: SSSOverrideUtils, user: str) -> None:
+        """
+        :param util: OverrideUser object.
+        :type util: SSSOverrideUser
+        :param user: User name.
+        :type user: str
+        """
+        self.util: SSSOverrideUtils = util
+        self.user: str = user
+
+    def add(
+        self,
+        *,
+        uid: int | None = None,
+        gid: int | None = None,
+        name: str | None = None,
+        home: str | None = None,
+        gecos: str | None = None,
+        shell: str | None = None,
+        certificate: str | None = None,
+    ) -> SSSOverrideUser:
+        """
+        Create new local override for user.
+
+        :param uid: User id, defaults to None
+        :type uid: int | None, optional
+        :param gid: Primary group id, defaults to None
+        :type gid: int | None, optional
+        :param name: | None, optional
+        :type name: str | None, optional
+        :param home: Home directory, defaults to None
+        :type home: str | None, optional
+        :param gecos: GECOS, defaults to None
+        :type gecos: str | None, optional
+        :param shell: Login shell, defaults to None
+        :type shell: str | None, optional
+        :param certificate: PKI certificate defaults to None
+        :type certificate: str | None, optional
+        :return: Self.
+        :rtype: SSSOverrideUser
+        """
+        args: CLIBuilderArgs = {
+            "name": (self.util.cli.option.VALUE, name),
+            "uid": (self.util.cli.option.VALUE, uid),
+            "gid": (self.util.cli.option.VALUE, gid),
+            "home": (self.util.cli.option.VALUE, home),
+            "gecos": (self.util.cli.option.VALUE, gecos),
+            "shell": (self.util.cli.option.VALUE, shell),
+            "certificate": (self.util.cli.option.VALUE, certificate),
+        }
+
+        self.util.logger.info(f"Creating local override for user {self.user} on {self.util.host.hostname}")
+        self.util.host.ssh.exec(
+            ["sss_override", "user-add", self.user] + self.util.cli.args(args), raise_on_error=False
```

Comment:
Why raise_on_error is False?

```
@pytest.mark.topology(KnownTopology.LDAP)
def test_override(client: Client, provider: GenericProvider):
    provider.user("tuser").add()
    client.sss_override.user("tuser").add(name="o-tuser", uid=50000, gid=60000)
    client.sssd.start()

    result = client.tools.id("o-tuser")
    print(str(result))
```

The test above does not work, because sss_overrride fails with:

```
INFO     2023-06-08 11:48:02,921 Executing command #11
                                 Host: client.test
                                 Shell: /usr/bin/bash -c
                                 User: root
                                 Command: sss_override user-add tuser --name o-tuser --uid 50000 --gid 60000
                                 Input: None
                                 Working directory: None
                                 Extra environment: 
INFO     2023-06-08 11:48:03,441 Previous command completed with exit code 1
                                 Output: 
                                 Error output: |
                                   [sssd] [confdb_expand_app_domains] (0x0010): No domains configured, fatal error!
                                   [sssd] [sss_tool_route] (0x0010): Command user-add does not handle initialization error [1432158246] No domain is enabled
```

But I didn't know it because no exception was risen.

Reply:
I should have removed it. I left 'raise_on_error=False' on the delete function. Right now, the SSSD teardown occurs before this, so this will always fail.

Reply:
I think you should also remove it from delete as it is not called during teardown anymore.